cmd/flags/admin.go

```diff
@@ -151,3 +154,27 @@ func (s *AdminServer) registerPprofHandlers() {
 func (s *AdminServer) Close() error {
 	return s.server.Shutdown(context.Background())
 }
+
+// URL return the admin servers URL
+// FIXME: Parsing the s.adminHostPort should already be done, why do it again? Valid port
+// detection and the like should be handled during the argument parsing step. But I cannot figure out
+// how that is done (or if). Then should we drop any pretence of checking the values?
+func (s *AdminServer) URL() (string, error) {
```

Comment:
Regardless of the decision made WRT parsing, `net.SplitHostPort` is probably a better choice than doing it manually.